src/sortStudents.ts

```diff
@@ -1,16 +1,72 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+const arrAvg = (arr:number[]): number => arr
+  .reduce((a, b) => a + b, 0) / arr.length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student [] {
+  const toSortStudents: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      toSortStudents.sort(
+        (currentStudent: Student,
+          nextStudent: Student): number => {
+          return (order === 'asc')
+            ? currentStudent[sortBy].localeCompare(nextStudent[sortBy])
+            : nextStudent[sortBy].localeCompare(currentStudent[sortBy]);
+        },
+      );
```

Comment:
fix you formatting everywhere
```suggestion
      toSortStudents.sort(
        (currentStudent: Student, nextStudent: Student): number => {
          return (order === 'asc')
            ? currentStudent[sortBy].localeCompare(nextStudent[sortBy])
            : nextStudent[sortBy].localeCompare(currentStudent[sortBy]);
        },
      );
```